includes/abstracts/abstract-wc-legacy-order.php

```diff
@@ -0,0 +1,615 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Legacy Abstract Order
+ *
+ * Legacy and deprecated functions are here to keep the WC_Abstract_Order clean.
+ * This class will be removed in future versions.
+ *
+ * @class       WC_Abstract_Order
+ * @version     2.6.0
+ * @package     WooCommerce/Classes
+ * @category    Class
+ * @author      WooThemes
+ */
+abstract class WC_Abstract_Legacy_Order {
+
+    /**
+     * Update a line item for the order.
+     *
+     * Note this does not update order totals.
+     *
+     * @since 2.2
+     * @param object|int $item order item ID or item object.
+     * @param WC_Product $product
+     * @param array $args data to update.
+     * @return int updated order item ID
+     */
+    public function update_product( $item, $product, $args ) {
+		_deprecated_function( 'WC_Order::update_product', '2.6', 'Interact with WC_Order_Item_Product class' );
+		if ( is_numeric( $item ) ) {
+            $item = $this->get_item( $item );
+        }
+        if ( ! is_object( $item ) || ! $item->is_type( 'line_item' ) ) {
+            return false;
+        }
+        if ( ! $this->get_id() ) {
+            $this->save(); // Order must exist
+        }
+
+		// BW compatibility with old args
+        if ( isset( $args['totals'] ) ) {
+            if ( isset( $args['totals']['subtotal'] ) ) {
+                $args['subtotal'] = $args['totals']['subtotal'];
+            }
+            if ( isset( $args['totals']['total'] ) ) {
+                $args['total'] = $args['totals']['total'];
+            }
+            if ( isset( $args['totals']['subtotal_tax'] ) ) {
+                $args['subtotal_tax'] = $args['totals']['subtotal_tax'];
+            }
+            if ( isset( $args['totals']['tax'] ) ) {
+                $args['total_tax'] = $args['totals']['tax'];
+            }
+            if ( isset( $args['totals']['tax_data'] ) ) {
+                $args['taxes'] = $args['totals']['tax_data'];
```

Comment:
These lines could probably be refactored/reduced down to a loop to check and set for each key.